src/erlcloud_util.erl

```diff
@@ -246,3 +247,20 @@ http_uri_encode(URI) ->
 http_uri_encode(URI) ->
     http_uri:encode(URI).
 -endif.
+
+-spec proplists_to_map(proplists:proplist()) -> map().
+proplists_to_map(Proplist) ->
+    proplists_to_map(Proplist, #{}).
+
+-spec proplists_to_map(proplists:proplist(), map()) -> map().
+proplists_to_map([], Acc) ->
+    Acc;
+proplists_to_map([{Key, Val} | Tail], Acc) when is_list(Val) ->
+    proplists_to_map(Tail, Acc#{Key => proplists_to_map(Val)});
```

Comment:
This looks potentially a little problematic on the ambiguity of array vs. proplist object. I think maybe `proplists_to_map/1` might need some guard in place to prevent it from doing the wrong thing? This was covered in [jsn:from_proplist](https://github.com/nalundgaard/jsn/blob/4fe1217740e00008db94c7c6e6bfcae4537e2705/src/jsn.erl#L864-L892), although that implementation leaves out the special `[{}]` representation for empty objects that jsx uses (see [note](https://github.com/nalundgaard/jsn/tree/4fe1217740e00008db94c7c6e6bfcae4537e2705#proplist-format-concerns)). I think it might be appropriate here to map `[{}]` -> `#{}`, `[]` -> `[]`, and evaluate non-empty lists based on whether they match the pattern `[{_,_}|_]` (convert to proplist to map) or not (convert each array element to map if they are objects).

Reply:
Thanks @nalundgaard . I've added special function clauses. Please double check